protos/flyteidl/service/dataproxy.proto

```diff
@@ -63,6 +64,45 @@ message CreateDownloadLocationResponse {
   google.protobuf.Timestamp expires_at = 2;
 }
 
+// ArtifactType
+enum ArtifactType {
+  // ARTIFACT_TYPE_UNDEFINED is the default, often invalid, value for the enum.
+  ARTIFACT_TYPE_UNDEFINED = 0;
+
+  // ARTIFACT_TYPE_DECK refers to the deck html file optionally generated after a task, a workflow or a launch plan
+  // finishes executing.
+  ARTIFACT_TYPE_DECK = 1;
+}
+
+message CreateDownloadLinkRequest {
+  oneof source {
+    // WorkflowId is the unique identifier for the workflow execution.
+    core.WorkflowExecutionIdentifier execution_id = 1;
+
+    // NodeId is the unique identifier for the node execution. For a task node, this will retrieve the output of the
+    // most recent attempt of the task.
+    core.NodeExecutionIdentifier node_id = 2;
+
+    // TaskId is the unique task execution identifier that points to a specific attempt of the task.
+    core.TaskExecutionIdentifier task_id = 3;
+  }
+
+  ArtifactType artifact_type = 4;
+
+  // ExpiresIn defines a requested expiration duration for the generated url. The request will be rejected if this
+  // exceeds the platform allowed max.
+  // +optional. The default value comes from a global config.
+  google.protobuf.Duration expires_in = 5;
+}
+
+message CreateDownloadLinkResponse {
+  // SignedUrl specifies the url to use to download content from (e.g. https://my-bucket.s3.amazonaws.com/randomstring/suffix.tar?X-...)
+  repeated string signed_url = 1;
```

Comment:
When does this return multiple values?

Reply:
Ketan commented that... in case we start using this to download FlyteDirectory in the future,... etc.